contracts/transfer/src/transfer.rs

```diff
@@ -96,6 +97,37 @@ impl TransferContract {
         Ok(())
     }
 
+    pub fn tx_hash(
+        nullifiers: &[BlsScalar],
+        outputs: &[Note],
+        anchor: &BlsScalar,
+        fee: &Fee,
+        crossover: Option<&Crossover>,
+        call: Option<&(ContractId, Transaction)>,
+    ) -> BlsScalar {
+        let mut hasher = Hasher::new();
+
+        nullifiers.iter().for_each(|n| hasher.update(n.to_bytes()));
+        outputs.iter().for_each(|o| hasher.update(o.to_bytes()));
+
+        hasher.update(anchor.to_bytes());
+        hasher.update(fee.to_bytes());
+
+        if let Some(c) = crossover {
+            hasher.update(c.to_bytes());
+        };
+
+        if let Some((cid, txdata)) = call {
+            hasher.update(cid.as_bytes());
+
+            // TODO review if as_bytes of `Transaction` is the same as canon
```

Comment:
`Canon` is no longer used to do this in `dusk-wallet-core`. As far as the wallet is concerned the transaction is just a `Vec<u8>`.

Reply:
Gotcha, removed the comment